src/diffusers/models/adapter.py

```diff
@@ -0,0 +1,246 @@
+from typing import List, Optional
+
+import torch
+import torch.nn as nn
+
+from ..configuration_utils import ConfigMixin, register_to_config
+from .modeling_utils import ModelMixin
+from .resnet import Downsample2D
+
+
+class BottleneckResnetBlock(nn.Module):
+    def __init__(self, in_c, mid_c, out_c, down, ksize=3, sk=False, use_conv=True, proj_ksize=1):
+        super().__init__()
+        ps = ksize // 2
+        proj_pad = proj_ksize // 2
+
+        if in_c != mid_c or sk is False:
+            self.in_conv = nn.Conv2d(in_c, mid_c, proj_ksize, 1, proj_pad)
+        else:
+            self.in_conv = None
+
+        if out_c != mid_c:
+            self.out_conv = nn.Conv2d(mid_c, out_c, proj_ksize, 1, proj_pad)
+        else:
+            self.out_conv = None
+
+        self.block1 = nn.Conv2d(mid_c, mid_c, 3, 1, 1)
+        self.act = nn.ReLU()
+        self.block2 = nn.Conv2d(mid_c, mid_c, ksize, 1, ps)
+
+        if sk is False:
+            self.skip = nn.Conv2d(in_c, mid_c, ksize, 1, ps)
+        else:
+            self.skip = None
+
+        self.down = down
+        if self.down is True:
+            self.down_opt = Downsample2D(in_c, use_conv=use_conv)
+
+    def forward(self, x):
+        if self.down is True:
+            x = self.down_opt(x)
+        if self.in_conv is not None:  # edit
+            x = self.in_conv(x)
+
+        h = self.block1(x)
+        h = self.act(h)
+        h = self.block2(h)
+        if self.skip is not None:
+            h = h + self.skip(x)
+        else:
+            h = h + x
+
+        if self.out_conv is not None:
+            h = self.out_conv(h)
+        return h
+
+
+class Adapter(ModelMixin, ConfigMixin):
```

Comment:
Super nit pick here: "Adapter" sounds really general but the adapter is specific to this model. Could we re-name it `T2IAdapter`? If more models end up using it and we "adapt" the adapter more, we could rename it to something more general